### PR TITLE
ReplaceLambdaWithInnerExpressionFix: use more descriptive text for partial applications

### PR DESCRIPTION
**Pull request by DedSec256**

### CODE REVIEW

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs [hunk 1]
```diff
@@ -12,6 +13,10 @@ type ReplaceLambdaWithInnerExpressionFix(warning: LambdaCanBeReplacedWithInnerEx
     override x.Text =
         match replaceCandidate with
         | :? IReferenceExpr as ref when isSimpleQualifiedName ref ->
-            sprintf "Replace lambda with '%s'" ref.QualifiedName
-        | :? IPrefixAppExpr -> "Replace lambda with partial application"
+            $"Replace lambda with '{ref.QualifiedName}'"
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs:16]: Is there a reason why a qualified name makes it better, even though the text may get considerably longer? Does it also properly work with all kind of arbitrary qualifier expressions?

File: ReSharper.FSharp/test/src/FSharp.Tests/QuickFixes/ReplaceLambdaWithInnerExpressionTest.fs [hunk 2]
```diff
@@ -25,6 +25,7 @@ type ReplaceLambdaWithInnerExpressionAvailabilityTest() =
 
     [<Test>] member x.``Text - Replace lambda with reference``() = x.DoNamedTest()
     [<Test>] member x.``Text - Replace lambda with qualified reference``() = x.DoNamedTest()
+    [<Test>] member x.``Text - Replace lambda with reference partial application``() = x.DoNamedTest()
```
Comment (auduchinok) [inline ReSharper.FSharp/test/src/FSharp.Tests/QuickFixes/ReplaceLambdaWithInnerExpressionTest.fs:28]: The names seem overcomplicated. We already know these tests are about lambdas, don't we?
```suggestion
    [<Test>] member x.``Text - Replace with reference partial application``() = x.DoNamedTest()
```

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs [hunk 3]
```diff
@@ -12,6 +13,10 @@ type ReplaceLambdaWithInnerExpressionFix(warning: LambdaCanBeReplacedWithInnerEx
...
+        | :? IPrefixAppExpr as app ->
+            match getRootFunctionExpr app with
+            | :? IReferenceExpr as ref when isSimpleQualifiedName ref ->
+                $"Replace with '{ref.QualifiedName}' partial application"
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs:20]: Is there a reason why a qualified name makes it better, even though the text may get considerably longer? Does it also properly work with all kind of arbitrary qualifier expressions?

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/PsiUtil.fs [hunk 4]
```diff
@@ -444,6 +444,11 @@ let rec tryFindRootPrefixAppWhereExpressionIsFunc ([<CanBeNull>] expr: IFSharpEx
         tryFindRootPrefixAppWhereExpressionIsFunc(prefixApp)
     else expr
 
+let rec getRootFunctionExpr ([<NotNull>] app : IPrefixAppExpr) =
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/PsiUtil.fs:447]: Any reason to have it in a common util? Are there other uses that could be replaced?

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs [hunk 5]
```diff
@@ -1,17 +1,34 @@
...
-            sprintf "Replace lambda with '%s'" ref.QualifiedName
-        | :? IPrefixAppExpr -> "Replace lambda with partial application"
+            $"Replace lambda with '{ref.QualifiedName}'"
+        | :? IPrefixAppExpr as app ->
+            match getRootFunctionExpr app with
+            | :? IReferenceExpr as ref ->
+                match ref.ShortName with
+                | name when name = SharedImplUtil.MISSING_DECLARATION_NAME ||
+                            PrettyNaming.IsOperatorName name ->
+                    $"Replace with '{ref.Reference.GetName()}' partial application"
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs:30]: Why `Replace with '???' partial application` is a good text for the case of the missing name?

Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs:30]: The operator case shouldn't be any special now after [RIDER-41848](https://youtrack.jetbrains.com/issue/RIDER-41848) is fixed.

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/PsiUtil.fs (hunk 4 above)
Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/PsiUtil.fs]: OK, now, when it was moved inside the quick fix, are you sure there are no existing uses that could benefit from having it in the util?

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs (hunk 3 above)
Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceLambdaWithInnerExpressionFix.fs]: No reply, but it's resolved.

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/PsiUtil.fs (hunk 4 above)
Comment (DedSec256) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/PsiUtil.fs]: Now I noticed that `IPrefixAppExpr` already has the property `InvokedReferenceExpression`, necessary for this (:

**Review by auduchinok — APPROVED** [review]

**Pull request merged**